lumicks/pylake/detail/widefield.py

```diff
@@ -327,3 +299,121 @@ def crop(self, roi):
     @property
     def shape(self):
         return self.y_max - self.y_min, self.x_max - self.x_min
+
+
+@dataclass
+class TransformMatrix:
+    """Wrapper around affine transformation matrix with convenience functions."""
+
+    matrix: np.ndarray = field(default=np.eye(3)[:2, :])
+
+    def __mul__(self, mat):
+        """Perform matrix multiplication such that `self * mat == np.matmul(self, mat)`."""
+        assert isinstance(mat, TransformMatrix), "Operands must be of type `TransformMatrix`."
+        m1 = np.vstack((self.matrix, [0, 0, 1]))
+        m2 = np.vstack((mat.matrix, [0, 0, 1]))
+        return TransformMatrix(np.matmul(m1, m2)[:2])
+
+    @classmethod
+    def with_offset(cls, alignment, x_offset, y_offset):
+        """Recalculate matrix with offset. Same implementation as Bluelake.
```

Comment:
I'd vote for `TransformMatrix(m).with_offset(x, y)`. Seems like the most logical and Python API to me.

Reply:
I originally suggested that as well, but that was before I looked closer and noticed that this operation does not do what I would expect from that name. I would prefer to be alerted that this is a nonstandard operation. `with_offset` makes me think it's just a translation of the operating point, which this is not.

The following produces a translation, which is not what I would expect as a user of this API:
```
rot = TransformMatrix.rotation(10, (10, 5))
irot = TransformMatrix.rotation(-10, (10, 5))
TransformMatrix.with_offset(rot.matrix, 10, 10) * TransformMatrix.with_offset(irot.matrix, 10, 10)
```

Reply:
I see. Then the dedicated `classmethod` makes sense. I'd suggest simply `from_alignment()` to avoid being too verbose. The offset parameters are clear in the signature and could be defaulted.